setup-ros.sh

```diff
@@ -27,15 +27,22 @@ apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 \
 
 echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" \
     > /etc/apt/sources.list.d/ros-latest.list
-echo "deb http://packages.ros.org/ros2/ubuntu $(lsb_release -sc) main" \
-    > /etc/apt/sources.list.d/ros2-latest.list
+# Add in ros2-testing to find the Foxy beta
+if [[ "$ROS_DISTRO" == "foxy" ]]; then
+	ros2_apt_source="ros2-testing"
+else
+	ros2_apt_source="ros2"
+fi
```

Comment:
This is mixing two different things, we can install from shadow-testing, and this is true for any distrib, or we can use a particular distro. Do you mind splitting those into two different arguments variables?

Reply:
Yeah we can do that - we'll be generating quite a few images though. Do you think we really need to be creating the `desktop` images? It seems like that's not a good image for CI (which is the primary intention here) since it includes so many dependencies that the user should be careful to include in their package.xml. I'm thinking maybe we pare it down to just building `ros-core` even

Reply:
I agree with how we're generating too many images. I second that we should skip building the `desktop` variant images, start building `ros-core` variant images, and maybe continue building the `ros-base` variant images. I also do think it is cleaner to separate out `ros2` and `ros2-testing` images.

Reply:
I don't disagree, I also think it's independent from this PR.